### PR TITLE
Change instances of HOMDEL and HETLOSS in frontend to DeepDel and ShallowDel

### PR DESCRIPTION
**Pull request by jtquach1**
Fix cBioPortal/cbioportal#6923 

Describe changes proposed in this pull request:

- Updated OQL parser grammar to support DeepDel and ShallowDel
  - Edited rules to use rules with their corresponding literals
  - Added tests for DeepDel and ShallowDel to represent HOMDEL and HETLOSS
  - Formatted pegjs file with [prettier](https://siefkenj.github.io/prettier-pegjs-playground/)
- Updated HOMDEL to DeepDel for CNA genes table
- Updated oncoprint tooltips

To do:
- Update subtitle next to gene on left oncoprint sidebar

Page tested: [link](http://localhost:3000/results/download?cancer_study_list=metastatic_solid_tumors_mich_2017%2Cchol_msk_2018&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&data_priority=0&profileFilter=0&case_set_id=all&gene_list=EP300%253A%2520DEEPDEL&geneset_list=%20&tab_index=tab_visualize&Action=Submit&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=chol_msk_2018_cna)

### CODE REVIEW

File: src/pages/studyView/StudyViewUtils.tsx [hunk 1]
```diff
@@ -1420,7 +1420,7 @@ export function getCNAColorByAlteration(
     alteration: string
 ): string | undefined {
     switch (alteration) {
-        case 'HOMDEL':
+        case 'DeepDel':
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx:1423]: would be nice to use the CNA_TO_ALTERATION constant here instead of plan string

**Review by adamabeshouse — APPROVED** [review]

**Comment by jtquach1** [comment]
HOMDEL is still present in 

the downloadable Excel sheet under the Oncoprint > Download tab:
image

and as a table row entry under "Type of Genetic Alterations Across All Samples":
image

I'll try to refactor code so HOMDEL->DeepDel, etc. will be cleaner. It looks like `homdel` and `hetloss` were used as object keys, hardcoded from the PegJS grammar values for the CNAType rule:
```
CNAType
        = "AMP"i { return "AMP"; }
        / "HOMDEL"i { return "HOMDEL"; }
        / "GAIN"i { return "GAIN"; }
        / "HETLOSS"i { return "HETLOSS"; }
```

File: src/shared/lib/oql/oql-parser.pegjs [hunk 2]
```diff
@@ -128,8 +128,10 @@ AnyTypeWithModifiersCommand
 
 CNAType
   = AMP { return "AMP"; }
-  / HOMDEL { return "HOMDEL"; }
+  / HOMDEL { return "DEEPDEL"; }
+  // / HOMDEL { return "HOMDEL"; }
```
Comment (adamabeshouse) [inline src/shared/lib/oql/oql-parser.pegjs]: please be sure to delete these comments before merging

**Comment by jtquach1** [comment]
Changes made, and why:


> HOMDEL is still present in
> 
> the downloadable Excel sheet under the Oncoprint > Download tab:
> image
> 
> and as a table row entry under "Type of Genetic Alterations Across All Samples":
> image
> 
> I'll try to refactor code so HOMDEL->DeepDel, etc. will be cleaner. It looks like `homdel` and `hetloss` were used as object keys in `isDatumWantedByOQLCNACommand()` from `oqlfilter.ts` and were hardcoded from the PegJS grammar values for the CNAType rule:
> 
> ```
> CNAType
>         = "AMP"i { return "AMP"; }
>         / "HOMDEL"i { return "HOMDEL"; }
>         / "GAIN"i { return "GAIN"; }
>         / "HETLOSS"i { return "HETLOSS"; }
> ```

The HOMDEL value was affected by the CNAType defined in the PegJS grammar

**Comment by jtquach1** [comment]
# Changes done, and why:

## [From Karthik's specification](https://github.com/cBioPortal/cbioportal/issues/6923#issue-538506451):
- [Study view CNA table](http://localhost:3000/study/summary?id=hnsc_tcga), from HOMDEL to DeepDel
image
- Oncoprint tooltip (hovering over the blip), from HOMODELETED to DeepDel
Before:
image

After:
image


## New changes
- Oncoprint tab

File: src/shared/lib/oql/oql-parser.d.ts [hunk 3]
```diff
@@ -20,22 +20,6 @@ export declare class SyntaxError {
 
 export declare function parse(str: string): OQLQuery | undefined;
 
-// start
```
Comment (jtquach1) [inline src/shared/lib/oql/oql-parser.d.ts:23]: These comments are outdated compared to the parser grammar (even before adding DEEPDEL to the CNAType rule), so I've removed them

File: src/shared/lib/oql/oql-parser.pegjs [hunk 4]
```diff
@@ -1,35 +1,58 @@
...
+  = "HOMDEL"i
+  / "DeepDel"i
+
 GAIN = "GAIN"i
-HETLOSS = "HETLOSS"i
+
+HETLOSS
+  = "HETLOSS"i
+  / "ShallowDel"i
+
```
Comment (jtquach1) [inline src/shared/lib/oql/oql-parser.pegjs:51]: These were the most significant changes in the OQL grammar, the rest of the changes are from an open-source prettier formatter for pegjs.

File: src/shared/lib/oql/oql-parser.pegjs [hunk 5]
```diff
@@ -54,134 +77,275 @@ PROT = "PROT"i
...
+  / cmd:PROTCommand { return cmd; }
+  / cmd:FUSIONCommand { return cmd; }
+  // MUT has to go at the end because it matches an arbitrary string at the end as a type of mutation
+  / cmd:MUTCommand { return cmd; }
 
 AnyTypeWithModifiersCommand
-    = d:DriverModifier !"_" { return {"alteration_type":"any", modifiers:[d]}; }
+  = d:DriverModifier !"_" { return { alteration_type: "any", modifiers: [d] }; }
 
 CNAType
```
Comment (jtquach1) [inline src/shared/lib/oql/oql-parser.pegjs:129]: e.g. here strings were replaced with the corresponding rules

**Comment by jtquach1** [comment]
Just did draft mode to run frontend PR off backend at https://github.com/cBioPortal/cbioportal/pull/8271, looks like localdb tests passed

File: env/custom.sh [hunk 6]
```diff
@@ -1,2 +1,3 @@
 # set e.g.
 # export CBIOPORTAL_URL="http://localhost:8080"
```
Comment (alisman) [inline env/custom.sh:2]: maybe take this out

File: env/master.sh [hunk 7]
```diff
@@ -1,2 +1,3 @@
-export CBIOPORTAL_URL="https://www.cbioportal.org"
+# export CBIOPORTAL_URL="https://www.cbioportal.org"
+export CBIOPORTAL_URL="http://localhost:8080"
```
Comment (alisman) [inline env/master.sh:2]: try not to commit this

**Comment by jtquach1** [comment]
Just reverted changes to `env/custom.sh` and `env/master.sh`

File: env/master.sh (hunk 7 above)
Comment (jtquach1) [reply env/master.sh]: reverted file

File: env/custom.sh (hunk 6 above)
Comment (jtquach1) [reply env/custom.sh]: reverted file

**Comment by alisman** [comment]
@inodb what's story here?

**Pull request closed without merging**